### PR TITLE
Updated: Use image cache for TADB images

### PR DESCRIPTION
**Pull request by danielunderwood**

### CODE REVIEW

File: lidarrmetadata/provider.py [hunk 1]
```diff
@@ -80,6 +81,18 @@ def _get_rate_limiter(key=None):
...
+def response_url(url: str) -> str:
+    """
+    Transforms a URL to a response URL, which can take into account things such as a hosted cache
+    for third-party services.
+    """
+    parsed = urlparse(url)
+    if parsed.netloc.endswith("theaudiodb.com"):
+        parsed = parsed._replace(
+            netloc=CONFIG.IMAGE_CACHE_HOST,
+            path=f"v1/tadb/{parsed.path}"
```
Comment (Qstick) [inline lidarrmetadata/provider.py]: Seems this is not working on testing, but testing the function local seem to produce a url with extra parameters. Additionally it's got two `//` after tadb

```
https://imagecache.lidarr.audio/v1/tadb//images/media/artist/banner/tvyypy1476729221.jpg
```

it should remove `images/media` to produce url below

```
https://imagecache.lidarr.audio/v1/tadb/artist/banner/tvyypy1476729221.jpg
```

**Comment by Qstick** [comment]
Also, Fixes https://github.com/Lidarr/Lidarr/issues/2182

**Comment by gold007eye** [comment]
Seems like this issue is back.  I have verified this on 2 installs (one existing Windows install (Version 2.2.4.4100 [nightly]), and 1 new Container (hotio) install Version 2.2.4.4100 [nightly]).   Going to the URL directly in browser works without issue, but in Lidarr anything trying to retrieve from theaudiodb.com gets 403 Forbidden:

````
[Warn] HttpClient: HTTP Error - Res: HTTP/2.0 [HEAD] https://www.theaudiodb.com/images/media/artist/thumb/xtstqw1393087453.jpg: 403.Forbidden (0 bytes)
 
[Warn] MediaCoverService: Couldn't download media cover for [00f82af1-6537-4d4b-9e03-103fffe19999][Baha Men]. HTTP request failed: [403:Forbidden] [HEAD] at [https://www.theaudiodb.com/images/media/artist/thumb/xtstqw1393087453.jpg]
````

I saw that this code was changed in 2022 to supposedly point to lidarr cache for "theaudiodb.com" urls, but not sure if something broke that isn't allowing this to happen anymore.

**Comment by gbolton2008** [comment]
I am also seeing this error on a fresh install of 2.2.3.4098. At the moment I seem to have been able to work around this by pointing lidarr at a squid proxy instance and replacing the User-Agent header with the string from Chrome which seems to be preventing theaudiodb.com from blocking requests

**Pull request merged**